Mage.Sets/src/mage/cards/o/OlorinsSearingLight.java

```diff
@@ -0,0 +1,99 @@
+
+        // Each opponent exiles a creature with the greatest power among creatures that player controls.
+        // Spell mastery -- If there are two or more instant and/or sorcery cards in your graveyard, Olorin's Searing Light deals damage to each opponent equal to the power of the creature they exiled.
+        this.getSpellAbility().addEffect(new OlorinsSearingLightEffect());
+    }
+
+    private OlorinsSearingLight(final OlorinsSearingLight card) {
+        super(card);
+    }
+
+    @Override
+    public OlorinsSearingLight copy() {
+        return new OlorinsSearingLight(this);
+    }
+}
+//See Crackling Doom
+class OlorinsSearingLightEffect extends OneShotEffect {
+
+    static FilterControlledCreaturePermanent filter = new FilterControlledCreaturePermanent("creature with the greatest power among creatures you control");
+    static {
+        filter.add(GreatestPowerControlledPredicate.instance);
+    }
+    public OlorinsSearingLightEffect() {
+        super(Outcome.Sacrifice);
+        this.staticText = "Each opponent exiles a creature with the greatest power among creatures that player controls.<br>"
+                +"<i>Spell mastery</i> &mdash; If there are two or more instant and/or sorcery cards in your graveyard, {this} deals damage to each opponent equal to the power of the creature they exiled.";
+    }
+
+    private OlorinsSearingLightEffect(final OlorinsSearingLightEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public OlorinsSearingLightEffect copy() {
+        return new OlorinsSearingLightEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null) {
+            List<Permanent> toExile = new ArrayList<>();
+            for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
+                if (controller.hasOpponent(playerId, game)) {
+                    Player opponent = game.getPlayer(playerId);
+                    if (opponent != null) {
+                        Target target = new TargetControlledCreaturePermanent(filter);
+                        target.withNotTarget(true);
+                        if (opponent.choose(outcome, target, source, game)) {
+                            Permanent permanentChosen = game.getPermanent(target.getFirstTarget());
+                            if (permanentChosen != null) {
+                                toExile.add(permanentChosen);
+                            }
+                        }
+                    }
+                }
+            }
+            for (Permanent permanent : toExile) {
+                Player opponent = game.getPlayer(permanent.getControllerId());
+                if (SpellMasteryCondition.instance.apply(game, source)){
+                    opponent.damage(permanent.getPower().getValue(),source, game);
```

Comment:
Very good point, fixed

Reply:
miss null check on player?

Reply:
I was thinking I didn't need to because I already did a null check earlier but you're right, there is time passing between the two which means I need to re-check.